src/auth/session/storage/redis.ts

```diff
@@ -85,13 +85,19 @@ export class RedisSessionStorage implements SessionStorage {
     const cleanShop = sanitizeShop(shop, true)!;
 
     const keys = await this.client.keys('*');
+
     const results: SessionInterface[] = [];
     for (const key of keys) {
       const rawResult = await this.client.get(key);
       if (!rawResult) continue;
 
+      try {
+
       const session = sessionFromEntries(JSON.parse(rawResult));
-      if (session.shop === cleanShop) results.push(session);
+      if (session?.shop === cleanShop) results.push(session);
+      } 
+      // do nothing if the rawResult is not a parse-able session
+      catch {}
```

Comment:
Should we catch the Json Error raised here or any more relevant parsing exception? I am conscious of hiding other exceptions that we may want to let through if they occur. I know this would require some custom code in order to get the type of the exception or re throw it if it is not what we want.
Not sure if this is very JS idiomatic but `catch all` blocks always scare me :)

Reply:
Narrowing error types isn't elegant in JS, but gave it go. I agree that the `catch all` is scary